### PR TITLE
 [ci skip] Update Guides to refer to main branch

### PR DESCRIPTION
**Pull request by hummusonrails**
### Summary

Given the recent change to `main` as the default branch for Rails, this pull request updates the documentation to refer to `main` instead of `master` where appropriate.

### CODE REVIEW

**Comment by MikeRogers0** [comment]
@bencgreenberg It looks like we've overlapped https://github.com/rails/rails/pull/41143 👍  

Do you want to merge my branch into this one? It looks like we both caught a few different instances :)

**Comment by bencgreenberg** [comment]
> @bencgreenberg It looks like we've overlapped #41143 👍
> 
> Do you want to merge my branch into this one? It looks like we both caught a few different instances of `master` which need updating :)

Great minds and all that I guess 😄 

Happy to do so!

**Comment by MikeRogers0** [comment]
> Great minds and all that I guess 😄
> 
> Happy to do so!

So true! Please do :) If merging is being annoying, you can also just run the commands I listed in my PR & commit those :)

**Comment by bencgreenberg** [comment]
> So true! Please do :) If merging is being annoying, you can also just run the commands I listed in my PR & commit the changes :)

I think I completed the merge successfully. Please check and let me know if I missed anything somehow! Thank you!

**Comment by MikeRogers0** [comment]
> I think I completed the merge successfully. Please check and let me know if I missed anything somehow! Thank you!

Looks perfect @bencgreenberg, nice one! I've closed my PR as it's no longer necessary :)

File: guides/source/contributing_to_ruby_on_rails.md [hunk 1]
```diff
@@ -39,13 +39,23 @@ Then, don't get your hopes up! Unless you have a "Code Red, Mission Critical, th
...
+* Template for Active Job issues: [gem](https://github.com/rails/rails/blob/main/guides/bug_report_templates/active_job_gem.rb) / [master](https://github.com/rails/rails/blob/main/guides/bug_report_templates/active_job_master.rb)
+* Template for Active Storage issues: [gem](https://github.com/rails/rails/blob/main/guides/bug_report_templates/active_storage_gem.rb) / [master](https://github.com/rails/rails/blob/main/guides/bug_report_templates/active_storage_master.rb)
+* Template for Action Mailbox issues: [gem](https://github.com/rails/rails/blob/main/guides/bug_report_templates/action_mailbox_gem.rb) / [master](https://github.com/rails/rails/blob/main/guides/bug_report_templates/action_mailbox_master.rb)
+* Generic template for other issues: [gem](https://github.com/rails/rails/blob/main/guides/bug_report_templates/generic_gem.rb) / [master](https://github.com/rails/rails/blob/main/guides/bug_report_templates/generic_master.rb)
+>>>>>>> otherrep/bug/references-to-master-branch
```
Comment (georgeclaghorn) [inline guides/source/contributing_to_ruby_on_rails.md:58]: Missed merge conflict resolution.

Comment (bencgreenberg) [reply guides/source/contributing_to_ruby_on_rails.md]: Thanks, @georgeclaghorn for catching it. Pushed up a resolution to it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -10,9 +10,9 @@
 * If you're unable to find an open issue addressing the problem, [open a new one](https://github.com/rails/rails/issues/new). Be sure to include a **title and clear description**, as much relevant information as possible, and a **code sample** or an **executable test case** demonstrating the expected behavior that is not occurring.
 
 * If possible, use the relevant bug report templates to create the issue. Simply copy the content of the appropriate template into a .rb file, make the necessary changes to demonstrate the issue, and **paste the content into the issue description**:
-  * [**Active Record** (models, database) issues](https://github.com/rails/rails/blob/master/guides/bug_report_templates/active_record_master.rb)
-  * [**Action Pack** (controllers, routing) issues](https://github.com/rails/rails/blob/master/guides/bug_report_templates/action_controller_master.rb)
-  * [**Generic template** for other issues](https://github.com/rails/rails/blob/master/guides/bug_report_templates/generic_master.rb)
+  * [**Active Record** (models, database) issues](https://github.com/rails/rails/blob/main/guides/bug_report_templates/active_record_master.rb)
```
Comment (yahonda) [inline CONTRIBUTING.md:13]: Would you consider to rename `action_controller_master.rb` to `action_controller_main.rb` ? Here these file names based on the master branch. It also applies to other *_master.rb files.

Comment (bencgreenberg) [reply CONTRIBUTING.md:13]: Hi @yahonda, I wasn't sure if that change should be included in this specific PR. I'd be happy to do that if it does.

Comment (yahonda) [reply CONTRIBUTING.md:13]: Looks like https://github.com/rails/rails/commit/077c66d5d6ef3a6f1cc54e4a431bfa5ea6831b97 already renamed the bug_report_templates file names.

Comment (bencgreenberg) [reply CONTRIBUTING.md]: Thank you, I just merged those changes into this branch.

Comment (bencgreenberg) [reply CONTRIBUTING.md]: Out of curiosity, was https://github.com/rails/rails/commit/077c66d5d6ef3a6f1cc54e4a431bfa5ea6831b97 a separate pull request that was opened previously before this one? I tried to find it, but I don't see it, or was it committed directly to the `main` branch absent a PR? As a first time contributor to the Rails docs, I'd like to know the process a bit more here. I had reviewed current PRs before opening this almost 5 days ago, and had not seen work in progress similar to it, so trying to understand how most of this was covered by that commit?

Comment (rafaelfranca) [reply CONTRIBUTING.md]: There was no pull request opened for this change. As this is an administrative change that required coordinated changes to the doc server I decide to make the changes myself so we don't get in a broken state.

**Pull request merged**